File: Adapters/Windows Phone/ISharedPreferences.cs

namespace DE.Myparcelservice.App.OSAPIInterfaces
{
	/// <summary>Created by Tilmann Stehle on 02.04.2015.</summary>
	public interface ISharedPreferences
	{
		void EditValue(string key, string value);

		void EditValue(string key, int value);

		void EditValue(string key, float value);

		string GetString(string key, string defaultValue);

		float GetFloat(string key, float defaultValue);

		int GetInt(string key, int defaultValue);

		bool GetBoolean(string key, bool defaultValue);

		void EditValue(string hasBoxesKey, bool hasBoxes);

		void Remove(string key);
	}
}

File: Adapters/Android/ISharedPreferences.java
package de.myparcelservice.app.OSAPIInterfaces;

/**
 * Created by Tilmann Stehle on 02.04.2015.
 */
public interface ISharedPreferences {
    void editValue(String key, String value);

    void editValue(String key, int value);

    void editValue(String key, float value);

    String getString(String key, String defaultValue);

    float getFloat(String key, Float defaultValue);
    int getInt(String key, int defaultValue);

    boolean getBoolean(String key, boolean defaultValue);

    void editValue(String hasBoxesKey, boolean hasBoxes);

    void remove(String key);
}
